### ISSUE TITLE
MeanHighPassFilter needs to be modernized post-floats

### ISSUE THREAD

**Issue by dganguli** [spacetx/starfish#590]
The following code in high_pass is no longer needed:

```
if image.dtype != np.uint16:
    DataFormatWarning( "Mean filters currently only support uint16 images. 
      Image data will be converted.")
image = img_as_uint(image)
```

FYI @kevinyamauchi

**Comment by kevinyamauchi** [comment]
Awesome! Should I take a crack at it to make sure I understand how types/conversions work now?

**Comment by ambrosejcarr** [comment]
@kevinyamauchi Did you get the information you needed from @dganguli post-sync yesterday? If not lets chat -- This would be great for you to contribute. 👍

**Comment by kevinyamauchi** [comment]
I think @dganguli and I are meeting today. I'll check in with him about it then.

**Comment by kevinyamauchi** [comment]
Fixed by #620
